src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -156,6 +157,8 @@ void metadata_dissemination_service::handle_leadership_notification(
                     std::move(ntp), term, lid);
               });
         });
+    }).handle_exception_type([](const ss::gate_closed_exception&) {
```

Comment:
Sorry, I think I was slightly wrong when I suggested handle_exception_type

In `with_gate`, it'll throw the exception synchronously (i.e. need a try/catch here).

Or, switch it to `try_with_gate` and it'll return an exceptional future and handle_exception_type can be used.

Personally I don't have a style preference between the two.

Reply:
Oh
I was confused by 
```
    (void)ss::with_gate(_gate, [this, tps = std::move(tps)]() mutable {
        return ss::do_with(
          std::move(tps),
          [this](const std::vector<model::topic_partition>& tps) {
              return cleanup_removed_topic_partitions(tps);
          });
    }).handle_exception([](std::exception_ptr e) {
        vlog(klog.warn, "Topic clean-up encountered error: {}", e);
    });
```
And did not see catch type exception.

Reply:
That code looks like it's relying on whatever is calling it to handle the gate exception (or it's buggy)